hangupsbot/plugins/telesync/__init__.py

```diff
@@ -698,13 +743,128 @@ def tg_command_get_me(bot, chat_id, args):
+
+        applicable_configurations = []
+        ho2tg_dict = self.configuration["memory"]["ho2tg"]
+        if conv_id in ho2tg_dict:
+            # combine config.json and memory options to generate a dict
+            config_clone = dict(self.configuration["config"])
+            config_clone.update({ self.configkey: [ ho2tg_dict[conv_id] ],
+                                  "hangouts": [ conv_id ] })
+            applicable_configurations.append({ "trigger": conv_id,
+                                               "config.json": config_clone })
+
+        return applicable_configurations
+
+    @asyncio.coroutine
+    def _send_to_external_chat(self, config, event):
+        conv_id = config["trigger"]
+        external_ids = config["config.json"][self.configkey]
+
+        user = event.passthru["original_request"]["user"]
+        message = event.passthru["original_request"]["message"]
+
+        """migrated from telesync _on_hangouts_message():
 
-    telesync_config = bot.config.get_by_path(['telesync'])
-    if not telesync_config['enabled']:
+        * by this point of execution, applicable_configuration() would have
+            already filtered only relevant events
+        * telesync configuration only allows 1-to-1 telegram-ho mappings, this
+            migrated function supports multiple telegram groups anyway"""
+
+        sync_text = message
+        photo_url = None
+
+        has_photo, photo_file_name = yield from _telesync_is_valid_image_link(sync_text)
+        if has_photo:
+            photo_url = sync_text
+            sync_text = "shared an image"
+
+        bridge_user = self._get_user_details(user, { "event": event })
+        username = bridge_user["preferred_name"]
+        if bridge_user["chat_id"]:
+            username = "<a href=\"https://plus.google.com/u/0/{}/about\">{}</a>".format( bridge_user["chat_id"],
+                                                                                         bridge_user["preferred_name"] )
+
+        chat_title = format(self.bot.conversations.get_name(conv_id))
+
+        if "chatbridge" in event.passthru and event.passthru["chatbridge"]["source_title"]:
+            chat_title = event.passthru["chatbridge"]["source_title"]
+
+        if "sync_chat_titles" not in config or config["sync_chat_titles"] and chat_title:
+            formatted_text = "{} ({}): {}".format( username,
+                                                   chat_title,
+                                                   sync_text )
+        else:
+            formatted_text = "{}: {}".format( username,
+                                              sync_text )
+
+        # send messages first
+        for eid in external_ids:
+            yield from tg_bot.sendMessage( eid,
+                                           formatted_text,
+                                           parse_mode = 'HTML',
+                                           disable_web_page_preview = True )
```

Comment:
fixed by 5b109b2 - there are certainly edge cases where bad html/formatting will cause errors like this since telegram api is so fussy about html